Sources/MapboxNavigation/NavigationViewportDataSource.swift

```diff
@@ -0,0 +1,424 @@
+import MapboxMaps
+import MapboxCoreNavigation
+import Turf
+import MapboxDirections
+
+/**
+ Class, which conforms to `ViewportDataSource` protocol and provides default implementation of it.
+ */
+public class NavigationViewportDataSource: ViewportDataSource {
+    
+    /**
+     Delegate, which is used to notify `NavigationCamera` regarding upcoming `CameraOptions`
+     related changes.
+     */
+    public var delegate: ViewportDataSourceDelegate?
+    
+    /**
+     `CameraOptions`, which are used on iOS when transitioning to `NavigationCameraState.following` or
+     for continuous updates when already in `NavigationCameraState.following` state.
+     */
+    public var followingMobileCamera: CameraOptions = CameraOptions()
+    
+    /**
+     `CameraOptions`, which are used on CarPlay when transitioning to `NavigationCameraState.following` or
+     for continuous updates when already in `NavigationCameraState.following` state.
+     */
+    public var followingCarPlayCamera: CameraOptions = CameraOptions()
+    
+    /**
+     `CameraOptions`, which are used on iOS when transitioning to `NavigationCameraState.overview` or
+     for continuous updates when already in `NavigationCameraState.overview` state.
+     */
+    public var overviewMobileCamera: CameraOptions = CameraOptions()
+    
+    /**
+     `CameraOptions`, which are used on CarPlay when transitioning to `NavigationCameraState.overview` or
+     for continuous updates when already in `NavigationCameraState.overview` state.
+     */
+    public var overviewCarPlayCamera: CameraOptions = CameraOptions()
+    
+    /**
+     Value of maximum pitch, which will be taken into account when preparing `CameraOptions` during
+     active guidance navigation.
+     
+     Defaults to `45.0` degrees.
+     */
+    public var maximumPitch: Double = 45.0
+    
+    /**
+     Altitude that the `NavigationCamera` initally defaults to when navigation starts.
+     
+     Defaults to `1000.0` meters.
+     */
+    public var defaultAltitude: CLLocationDistance = 1000.0
+    
+    /**
+     Controls the distance on route after the current maneuver to include in the frame.
+     
+     Defaults to `100.0` meters.
+     */
+    public var distanceToFrameAfterManeuver: CLLocationDistance = 100.0
+    
+    /**
+     Controls how much the bearing can deviate from the location's bearing, in degrees.
+     
+     In case if set, the `bearing` property of `CameraOptions` during active guidance navigation
+     won't exactly reflect the bearing returned by the location, but will also be affected by the
+     direction to the upcoming framed geometry, to maximize the viewable area.
+     
+     Defaults to `20.0` degrees.
+     */
+    public var maximumBearingSmoothingAngle: CLLocationDirection? = 20.0
```

Comment:
As per our discussion with @d-prukop `distanceToFrameAfterManeuver` and `maximumBearingSmoothingAngle` were made publicly available to provide more control over default camera behavior. There are several other properties, which are exposed in Mapbox Navigation for Android, but haven't made their way into `main` branch yet (and not yet present on iOS). We can add those in scope of tail-work. iOS implementation also doesn't use boolean flags to control usage of such properties (to simplify API), instead certain parameters provided as optionals.